common/src/main/java/com/microsoft/identity/common/internal/ui/webview/certbasedauth/NfcSmartcardCertBasedAuthChallengeHandler.java

```diff
@@ -51,14 +53,42 @@ public NfcSmartcardCertBasedAuthChallengeHandler(@NonNull final Activity activit
     }
 
     /**
-     * Pauses smartcard discovery, if the particular authentication method isn't meant to have
-     *  discovery active throughout the entire flow.
+     * To be called when user interaction is needed, or to prepare for any unexpected user interaction.
+     * When a connection is no longer actively being used, the dialog flow should pause
+     * so the user can remove their smartcard before flow can continue.
+     * @param nextInteractionCallback the next logic to be run after a user removes their smartcard.
      */
     @Override
-    protected void onPausedSmartcardDiscovery() {
-        //Helps prevent unnecessary callback trigger. Nfc discovery should only be active when
-        // the user is prompted to tap.
-        mCbaManager.stopDiscovery(mActivity);
+    protected void prepForNextUserInteraction(@Nullable final IDisconnectionCallback nextInteractionCallback) {
```

Comment:
Should the callback rather be a NonNull?

Reply:
Yep, there's no reason for it to be Nullable. I think it was residual from a previous implementation. I'll change it to NonNull.